test/test_op_aliases.py

```diff
@@ -0,0 +1,144 @@
+import torch
+from torch.testing import FileCheck
+
+from torch.testing._internal.common_utils import \
+    (run_tests)
+from torch.testing._internal.jit_utils import JitTestCase
+from torch.testing._internal.common_device_type import \
+    (instantiate_device_type_tests, skipCPUIfNoLapack, skipCUDAIfNoMagma, onlyCPU)
+
+# Information for generating an alias test
+# NOTE: ending the alias_name with an underscore will interpret the test
+#   as the test for an inplace method of that name
+class AliasInfo(object):
+    __slots__ = ['alias_name', 'alias_op', 'original_name', 'original_op',
+                 'get_input', 'get_args', 'decorators']
+
+    def __init__(self,
+                 alias_name,  # the name of the alias
+                 alias_op,  # the aliased op
+                 original_name,  # the name of the original function
+                 original_op,  # the original op
+                 get_input,  # callable (device)->tensor that returns the first tensor argument
+                 *,
+                 get_args=lambda d: (),  # callable (device)->tuple that returns additional positional arguments
+                 decorators=()):  # decorators to apply to the test
+        self.alias_name = alias_name
+        self.alias_op = alias_op
+        self.original_name = original_name
+        self.original_op = original_op
+        self.get_input = get_input
+        self.get_args = get_args
+        self.decorators = decorators
+
+alias_infos = (
+    AliasInfo('absolute', torch.absolute, 'abs', torch.abs,
+              lambda d: torch.randn(20, device=d) - .5),
+    AliasInfo('absolute_', torch.Tensor.absolute_, 'abs_', torch.Tensor.abs_,
+              lambda d: torch.randn(20, device=d) - .5),
+    AliasInfo('clip', torch.clip, 'clamp', torch.clamp,
+              lambda d: torch.randn(20, device=d), get_args=lambda d: (.4, .6)),
+    AliasInfo('clip_', torch.Tensor.clip_, 'clamp_', torch.Tensor.clamp_,
+              lambda d: torch.randn(20, device=d), get_args=lambda d: (.4, .6)),
+    AliasInfo('linalg.det', torch.linalg.det, 'det', torch.det,
+              lambda d: torch.randn(10, 10, device=d),
+              decorators=(skipCPUIfNoLapack, skipCUDAIfNoMagma)),
+    # NOTE: only runs on CPU because it leaks CUDA memory
+    #   (see https://github.com/pytorch/pytorch/issues/43119)
+    AliasInfo('outer', torch.outer, 'ger', torch.ger,
+              lambda d: torch.randn(20, device=d), get_args=lambda d: (torch.randn(20, device=d),),
+              decorators=(onlyCPU,)),
+    AliasInfo('arccosh', torch.arccosh, 'acosh', torch.acosh,
```

Comment:
it's not important, but are you sure you want to test almost only nans here?

Reply:
I was also thinking about this. I'll just add 2 to the input tensor so we're in the domain

Reply:
mostly in the domain :-)